src/main.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	. "screepsgo/src/screeps"
+	"strconv"
+	"syscall/js"
+)
+
+//export Loop
+func Loop() {
+	var spawn = Game.Spawns()["Spawn1"]
+	var room = spawn.Room()
+
+	var myStructures = room.Find(FIND_MY_STRUCTURES)
+	var tower *StructureTower = nil
+	for _, v := range myStructures {
+		var structure = Structure{RoomObject: v} // TODO
+		if structure.StructureType() == STRUCTURE_TOWER {
+			tower = &StructureTower{OwnedStructure: OwnedStructure{Structure: structure}} // TODO
+			break
+		}
+	}
+	if tower != nil {
+		var structures = tower.Room().Find(FIND_STRUCTURES)
+		var damagedStructures = []RoomObject{} // TODO
+		for _, v := range structures {
+			var structure = Structure{RoomObject: v} // TODO
+			if structure.Hits() != nil && *structure.Hits() < *structure.HitsMax() {
+				damagedStructures = append(damagedStructures, v)
+			}
+		}
+
+		var closestDamagedStructure = tower.Pos().FindClosestByRange_Objects(damagedStructures)
+		if closestDamagedStructure != nil {
+			tower.Repair(Structure{RoomObject: *closestDamagedStructure}) // TODO
+		}
+
+		var closestHostile = tower.Pos().FindClosestByRange(FIND_HOSTILE_CREEPS)
+		if closestHostile != nil {
+			tower.Attack(Creep{RoomObject: *closestHostile})
+		}
+	}
+
+	var harvesters = 0
+	var upgraders = 0
+	var builders = 0
+	for _, creep := range Game.Creeps() {
+		var role = creep.Memory().Get("role")
+		if role.IsUndefined() {
+			continue
+		}
+
+		if creep.Memory().Get("role").String() == "harvester" {

```

Comment:
It would be good to have these strings as constants in the screeps package. We risk typos every time we do it by hand.

Reply:
"harvester" is not API related. This is bot logic, as is written in tutorial bot code, which is kinda bad on purpose.